lib/public/services/detectors/detectorsProvider.js

```diff
@@ -43,14 +52,54 @@ export class DetectorsProvider extends Observable {
     }
 
     /**
-     * Returns the list of all the physical available detectors
-     * E.g `TST` detector is not phsycial system so it is no included in the results
+     * Return all detectors list observable data
+     *
+     * @return {ObservableData<RemoteData<Detector[], ApiError>>} the observable detectors list
+     */
+    get all$() {
+        if (this._isStale()) {
+            this._load();
+        }
+        return this._all$;
+    }
+
+    /**
+     * Return physical (meaning actual detectors, for example excluding TST) detectors list observable data
+     *
+     * @return {ObservableData<RemoteData<Detector[], ApiError>>} the observable physical detectors list
+     */
+    get physical$() {
+        if (this._isStale()) {
+            this._load();
+        }
+        return this._physical$;
+    }
+
+    /**
+     * States if the cached detectors list need to be created or updated
+     *
+     * @return {boolean} true if the detectors list must be refreshed
+     * @private
+     */
+    _isStale() {
+        return this._all$.getCurrent().match({
+            NotAsked: () => true,
+            Other: () => false,
```

Comment:
No reload if failed?

Reply:
For now I just put the same behavior, refactoring only

Reply:
Simply refactoring, not changing the behavior